sql/core/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveSessionCatalog.scala

```diff
@@ -449,8 +454,17 @@ class ResolveSessionCatalog(
         partitionSpec)
 
     case ShowColumnsStatement(tbl, ns) =>
+      if (ns.isDefined && ns.get.length > 1) {
+        throw new AnalysisException(
+          s"Namespace name should have only one part if specified: ${ns.get.quoted}")
+      }
+      val nameParts = if (ns.isDefined && tbl.length == 1) {
```

Comment:
Added.

Reply:
Actually, the part confuses me is when would `tbl.length != 1`. Is it because sometimes we would keep the namespace in the table name? It just feels strange that ns might not represent the actual namespace for the table.

Reply:
Yea, you can do the following: `SHOW COLUMNS FROM db1.t1 IN db1`